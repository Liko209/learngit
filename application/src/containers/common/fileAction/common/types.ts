import FileItemModel from '@/store/models/FileItem';

/*
 * @Author: looper Wang (looper.wang@ringcentral.com)
 * @Date: 2019-05-30 09:40:39
 * Copyright Â© RingCentral. All rights reserved.
 */

export interface IFileActionBaseViewModel {
  fileId: number;
  item: FileItemModel;
  fileName: string;
}

export type FileActionProps = {
  fileId: number;
  postId?: number;
};
