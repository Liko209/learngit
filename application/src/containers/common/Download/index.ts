/*
 * @Author: Chris Zhan (chris.zhan@ringcentral.com)
 * @Date: 2019-03-12 10:23:36
 * Copyright Â© RingCentral. All rights reserved.
 */
export * from './Download';
