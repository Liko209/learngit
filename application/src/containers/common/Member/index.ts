/*
 * @Author: Aaron Huo (aaron.huo@ringcentral.com)
 * @Date: 2019-03-11 16:25:35,
 * Copyright Â© RingCentral. All rights reserved.
 */

export { default as Member } from './Member';
