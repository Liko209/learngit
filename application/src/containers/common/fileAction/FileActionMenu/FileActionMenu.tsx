/*
 * @Author: wayne.zhou
 * @Date: 2019-05-27 14:31:37
 * Copyright Â© RingCentral. All rights reserved.
 */
import React, { Component } from 'react';
import { JuiIconButton, IconButtonVariant } from 'jui/components/Buttons';
import { FileDeleteAction, FileDeleteActionProps } from '../FileDeleteAction';
import {
  FileNameEditAction,
  FileNameEditActionProps,
} from '../FileNameEditAction';
import { ViewInPostAction } from '../ViewInPostAction';
import { JuiMenuList } from 'jui/components/Menus';
import { JuiPopperMenu, AnchorProps } from 'jui/pattern/PopperMenu';
import { WithTranslation, withTranslation } from 'react-i18next';

type FileActionMenuProps = {
  fileId: number;
  postId?: number;
  disablePortal?: boolean;
  variant?: IconButtonVariant;
  showViewInPostAction?: boolean;
  groupId?: number;
  asyncOperationDecorator?: FunctionDecorator;
} & FileDeleteActionProps &
  FileNameEditActionProps &
  WithTranslation;

type State = { open: boolean; anchorEl: EventTarget | null };
class InnerComponent extends Component<FileActionMenuProps, State> {
  static defaultProps: Partial<FileActionMenuProps> = {
    variant: 'plain',
    showViewInPostAction: false,
  };

  state = {
    open: false,
    anchorEl: null,
  };

  private _Anchor = ({ tooltipForceHide }: AnchorProps) => {
    const { t, variant } = this.props;
    return (
      <JuiIconButton
        size='medium'
        ariaLabel={t('common.more')}
        variant={variant}
        data-test-automation-id='fileActionMore'
        tooltipTitle={t('common.more')}
        onClick={this.openPopper}
        tooltipForceHide={tooltipForceHide}
      >
        more_vert
      </JuiIconButton>
    );
  };

  openPopper = (evt: React.MouseEvent) => {
    const { currentTarget } = evt;
    this.setState(state => ({
      anchorEl: currentTarget,
      open: !state.open,
    }));
  };

  closePopper = () => {
    this.setState({
      open: false,
    });
  };
  render() {
    const {
      showViewInPostAction,
      fileId,
      postId,
      groupId,
      disablePortal,
      asyncOperationDecorator,
      ...rest
    } = this.props;
    return (
      <JuiPopperMenu
        Anchor={this._Anchor}
        open={this.state.open}
        anchorEl={this.state.anchorEl}
        onClose={this.closePopper}
        disablePortal={disablePortal}
        data-test-automation-id={'fileActionMenu'}
      >
        <JuiMenuList data-test-automation-id={'fileActionMenuList'}>
          <FileNameEditAction fileId={fileId} postId={postId} {...rest} />
          {showViewInPostAction && groupId && (
            <ViewInPostAction
              asyncOperationDecorator={asyncOperationDecorator}
              fileId={fileId}
              groupId={groupId}
            />
          )}
          <FileDeleteAction fileId={fileId} postId={postId} {...rest} />
        </JuiMenuList>
      </JuiPopperMenu>
    );
  }
}

const FileActionMenu = withTranslation('translations')(InnerComponent);

export { FileActionMenu, FileActionMenuProps };
