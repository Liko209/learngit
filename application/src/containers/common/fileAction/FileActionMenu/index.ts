/*
 * @Author: wayne.zhou
 * @Date: 2019-05-27 17:47:52
 * Copyright Â© RingCentral. All rights reserved.
 */

export * from './FileActionMenu';
