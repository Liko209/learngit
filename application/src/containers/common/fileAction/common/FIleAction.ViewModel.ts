/*
 * @Author: looper Wang (looper.wang@ringcentral.com)
 * @Date: 2019-05-30 09:40:39
 * Copyright Â© RingCentral. All rights reserved.
 */

import { Post } from 'sdk/module/post/entity';
import PostModel from '@/store/models/Post';
import { computed } from 'mobx';
import { Item } from 'sdk/module/item/entity';
import FileItemModel from '@/store/models/FileItem';
import { getEntity } from '@/store/utils';
import { ENTITY_NAME } from '@/store';
import { StoreViewModel } from '@/store/ViewModel';
import { FileActionProps } from './types';

class FileActionViewModel extends StoreViewModel<FileActionProps> {
  @computed
  private get _id() {
    return this.props.fileId;
  }

  @computed
  get item() {
    return getEntity<Item, FileItemModel>(ENTITY_NAME.ITEM, this._id);
  }

  @computed
  get fileName() {
    return this.item.name;
  }

  @computed
  get post() {
    const { postId } = this.props;
    if (postId) {
      return getEntity<Post, PostModel>(ENTITY_NAME.POST, postId);
    }
    return null;
  }

  @computed
  get versionIndex() {
    const versions = this.item.versions;
    if (this.post) {
      if (this.post.itemData && versions) {
        const itemData = this.post.itemData;
        if (
          itemData &&
          itemData.version_map &&
          itemData.version_map[this._id]
        ) {
          const version = itemData.version_map[this._id];
          return versions.length - version;
        }
      }
      return versions.length - 1;
    }
    // There is no post data in the right rail
    return 0;
  }
}

export { FileActionViewModel };
