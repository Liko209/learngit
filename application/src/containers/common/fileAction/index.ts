/*
 * @Author: wayne.zhou
 * @Date: 2019-05-27 17:48:00
 * Copyright Â© RingCentral. All rights reserved.
 */

export * from './FileDeleteAction';
export * from './FileActionMenu';
