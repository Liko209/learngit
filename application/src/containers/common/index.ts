/*
 * @Author: Devin Lin (devin.lin@ringcentral.com)
 * @Date: 2018-11-23 15:49:38
 * Copyright Â© RingCentral. All rights reserved.
 */

export * from './Favorite';
export * from './Privacy';
export * from './PreloadImg';
