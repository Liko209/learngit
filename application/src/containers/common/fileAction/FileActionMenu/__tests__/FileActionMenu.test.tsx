import { FileActionMenuView } from '../FileActionMenu';
import { getFileType } from '@/common/getFileType';
import { shallow } from 'enzyme';
import React from 'react';
import { FileType } from '@/store/models/FileItem';

jest.mock('@/common/getFileType');

describe('FileActionMenu', () => {
  it('should render correctly', () => {
    getFileType = jest.fn().mockReturnValue({ type: FileType.image });
    const wrapper = shallow(
      <FileActionMenuView
        fileId={1}
        scene="conversationHistory"
        item={{} as any}
        fileName={'filename'}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should not render viewInBrowser when item is not image', () => {
    getFileType = jest.fn().mockReturnValue({ type: '' });
    const wrapper = shallow(
      <FileActionMenuView
        fileId={1}
        scene="conversationHistory"
        item={{} as any}
        fileName={'filename'}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
