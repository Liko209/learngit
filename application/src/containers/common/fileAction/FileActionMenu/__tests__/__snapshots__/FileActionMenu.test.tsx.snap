// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileActionMenu should not render viewInBrowser when item is not image 1`] = `
<JuiPopperMenu
  Anchor={[Function]}
  anchorEl={null}
  data-test-automation-id="fileActionMenu"
  onClose={[Function]}
  open={false}
>
  <JuiMenuList
    data-test-automation-id="fileActionMenuList"
  >
    <Container(FileNameEditActionViewComponent)
      fileId={1}
      fileName="filename"
      item={Object {}}
      t={[Function]}
      variant="plain"
    />
    <Container(FileDeleteActionViewComponent)
      fileId={1}
      fileName="filename"
      item={Object {}}
      t={[Function]}
      variant="plain"
    />
  </JuiMenuList>
</JuiPopperMenu>
`;

exports[`FileActionMenu should render correctly 1`] = `
<JuiPopperMenu
  Anchor={[Function]}
  anchorEl={null}
  data-test-automation-id="fileActionMenu"
  onClose={[Function]}
  open={false}
>
  <JuiMenuList
    data-test-automation-id="fileActionMenuList"
  >
    <ViewInBrowserTabAction
      item={Object {}}
      scene="conversationHistory"
    />
    <Container(FileNameEditActionViewComponent)
      fileId={1}
      fileName="filename"
      item={Object {}}
      t={[Function]}
      variant="plain"
    />
    <Container(FileDeleteActionViewComponent)
      fileId={1}
      fileName="filename"
      item={Object {}}
      t={[Function]}
      variant="plain"
    />
  </JuiMenuList>
</JuiPopperMenu>
`;
